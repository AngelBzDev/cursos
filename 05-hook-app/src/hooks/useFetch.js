import { useEffect, useRef, useState } from 'react'

export const useFetch = (url) => {
   
   const isMouted = useRef(true);

   const [state, setState] = useState({
      data: null, 
      loading: true,
      error: null
   })

   useEffect(() => {
      return () => {
         isMouted.current = false;
      }
   }, []);

   useEffect(() => {

      fetch(url)
         .then(response => response.json())
         .then(data => {            
            if(isMouted.current) {
               setState({ 
                  loading: false, 
                  error: null, 
                  data: data 
               });
            }
         });
         
   }, [url]);

   return state;

}
