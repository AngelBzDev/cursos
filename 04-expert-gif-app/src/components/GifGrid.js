import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem'

export const GifGrid = ({ category }) => {

   const { data:images, loading } = useFetchGifs( category );

   

   return (
      <>
         <h3 className="category animate__animated animate__fadeInRight">{ category }</h3>

         { loading && <p>Cargando...</p> }
         
         <div className="card-container animate__animated animate__fadeInLeft">
         { images.map( img  => (
               <GifGridItem 
                  key={ img.id }
                  { ...img }
               />
            ))
         }
         </div>
      </>
   )
}

Type TabNine::sem to enable semantic completion for JavaScript.
This will run the following commands:
Type TabNine::no_sem to suppress this message.
   To learn more about semantic completion, see https://tabnine.com/semantic.


GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}

