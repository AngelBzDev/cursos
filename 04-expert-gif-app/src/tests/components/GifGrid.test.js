import { shallow } from 'enzyme';  
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componente GifGrid', () => {
   
   const category = 'Pitufos';
   /* test('Debe de mostrar correctemente el componente', () => {
      //const wrapper = shallow(<GifGrid category={category}/>);
      expect(wrapper).toMatchSnapshot();
   }) */

   test('debe de mostrar items cuando se cargan imagenes useFetchGifts', () => {
      const gifs = [{
         id: '12ji',
         url: 'https://localhost/xdxdxd'
      }]

      useFetchGifs.mockReturnValue({
         data: gifs,
         loading: true
      })

      const wrapper = shallow(<GifGrid category={category}/>)
      
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('p').exists()).toBe(true);
      expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

   })
   
})
