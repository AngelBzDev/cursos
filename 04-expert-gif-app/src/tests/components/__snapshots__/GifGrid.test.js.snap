// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid debe de mostrar items cuando se cargan imagenes useFetchGifts 1`] = `
<Fragment>
  <h3
    className="category animate__animated animate__fadeInRight"
  >
    Pitufos
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-container animate__animated animate__fadeInLeft"
  >
    <GifGridItem
      id="12ji"
      key="12ji"
      url="https://localhost/xdxdxd"
    />
  </div>
</Fragment>
`;
